query/regexp.go

```diff
@@ -43,20 +43,20 @@ func LowerRegexp(r *syntax.Regexp) *syntax.Regexp {
 	return &newRE
 }
 
-// optimizeRegexp converts capturing groups to non-capturing groups.
+// OptimizeRegexp converts capturing groups to non-capturing groups.
 // Returns original input if an error is encountered
-func optimizeRegexp(re *syntax.Regexp) *syntax.Regexp {
-	r := convertCapture(re)
+func OptimizeRegexp(re *syntax.Regexp, flags syntax.Flags) *syntax.Regexp {
```

Comment:
In Sourcegraph, do you plan to call `OptimizeRegexp` directly or only indirectly via RegexpQuery?